File: bindings/python/examples/how_tos/simple_transaction/request_funds.py
import os

from dotenv import load_dotenv
from iota_sdk import Wallet, WalletOptions

# This example requests funds from the faucet

load_dotenv()

FAUCET_URL = os.environ.get(
    'FAUCET_URL',
    'https://faucet.testnet.shimmer.network/api/enqueue')

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))

address = wallet.address()
print(address)

response = wallet.get_client().request_funds_from_faucet(FAUCET_URL, address=address)
print(response)

File: bindings/nodejs/examples/how_tos/simple_transaction/request-funds.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/simple_transaction/request-funds.ts

// This example requests funds from the faucet
async function run() {
    initLogger();
    for (const envVar of ['FAUCET_URL', 'WALLET_DB_PATH']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is not defined`);
        }
    }
    try {
        const faucetUrl = process.env.FAUCET_URL as string;

        // Create the wallet
        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        const address = await wallet.address();
        console.log(address);

        const client = await wallet.getClient();
        const faucetResponse = await client.requestFundsFromFaucet(
            faucetUrl,
            address,
        );
        console.log(faucetResponse);
    } catch (error) {
        console.error('Error: ', error);
    }
}

void run().then(() => process.exit());
